src/main/java/no/unit/nva/institution/proxy/HttpExecutorImpl.java

```diff
@@ -85,21 +84,30 @@ public JsonNode getNestedInstitution(URI uri, Language language)
         String name = MapUtils.getNameValue(institutionUnit.getName());
         NestedInstitutionGenerator generator = new NestedInstitutionGenerator();
         generator.setInstitution(unitUri, name);
+
```

Comment:
I'd restructure the code so that failures occur before object initialization

`
        URI unitUri = getInstitutionUnitUri(uri, language);
        InstitutionBaseDto institutionUnit = getInstitutionBaseDto(unitUri, language);

        List<URI> unitUris = getUnitUris(institutionUnit.getId(), language);
        List<Try<SubSubUnitDto>> subSubUnitDtoResponses = fetchSubSubUnitInformation(language, unitUris);
        multipleRequestFailures(subSubUnitDtoResponses);

        NestedInstitutionGenerator generator = new NestedInstitutionGenerator();
        String name = MapUtils.getNameValue(institutionUnit.getName());
        generator.setInstitution(unitUri, name);
        addSubSubUnitsToNestedInstitutionGenerator(generator, subSubUnitDtoResponses);

        return generator.getNestedInstitution();
`

Reply:
I guess that setInstitution() can just take the InstitutionUnitDto and name and URI can be got there